### PR TITLE
update french regions (departements)

### PR DESCRIPTION
**Pull request by npracht**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅ 
| New feature/enhancement? (use the a.x branch)      | 🟥 
| Deprecations?                          | 🟥 
| BC breaks? (use the c.x branch)        | ✅ 
| Automated tests included?              | Not relevant
| Related user documentation PR URL      | Not relevant
| Related developer documentation PR URL | Not relevant
| Issue(s) addressed                     | 

#### Description:
This PR updates the list of French region relying on ISO_3166-2. This fixes 3 current issues:
1. Add missing regions
2. Update oversea french regions (add and rename one properly)
3. Reorder them in ISO order (region number instead of alphabetical)

#### Steps to test this PR:
* Check list if same as ISO_3166-2
* Check dropdown list of region is still working (in contact creation page for example)

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10802"></a>

### CODE REVIEW

File: app/bundles/CoreBundle/Assets/json/regions.json [hunk 1]
```diff
@@ -925,101 +925,107 @@
...
 		"Landes",
 		"Loir-et-Cher",
 		"Loire",
+		"Haute-Loire",
 		"Loire-Atlantique",
 		"Loiret",
 		"Lot",
 		"Lot-et-Garonne",
-		"Loz\u00e8re",
+		"Lozère",
```
Comment (escopecz) [inline app/bundles/CoreBundle/Assets/json/regions.json:976]: This has 1 disadvantage which is that if this value is already selected by a contact then the information will be lost with this change. Ideally, this PR should have a migration that would find and replace the old version to the new one.

**Review by escopecz — COMMENTED** [review]
Any reason why not use the alphabetical order? I can see some of the regions are just moved to different place in the list.

File: app/bundles/CoreBundle/Assets/json/regions.json (hunk 1 above)
Comment (kuzmany) [reply app/bundles/CoreBundle/Assets/json/regions.json:976]: @escopecz butI checked it and `Loz\u00e8re` was stored to DB  as `Lozère`. These kind of regions could break something. Do you see something?

Comment (escopecz) [reply app/bundles/CoreBundle/Assets/json/regions.json:976]: I see. In that case there is no need for a migration.

**Comment by kuzmany** [comment]
@escopecz migrations added for two exact changes 

Val d'Oise --> Val-d'Oise
Réunion --> La Réunion

**Review by escopecz — DISMISSED** [review]
Thanks! It looks good from the code perspective. The migration executed successfully. I can't tell whether the French regions are correct but since Norman is from there I believe it is 👍

**Comment by npracht** [comment]
You would win visiting our regions :P
More seriously i simply followed the ISO_3166-2.

**Comment by escopecz** [comment]
I couldn't be trusted doing this for Czech Republic. I'd be tempted to rename some regions to more fun alternatives ;)

**Comment by RCheesley** [comment]
@npracht this has the bc-break label but I think that might be accidental? Or should it be a 5.x PR?

**Comment by npracht** [comment]
@kuzmany i think there is BC. People sending or fetching previous wrong value by API will not work anymore ?

**Comment by escopecz** [comment]
I'm adding this to the milestone for Mautic 5 then. Please also document the BC change in the UPGRADING doc. Once that is done we can merge it into the 5.x branch

**Comment by kuzmany** [comment]
@escopecz I rebased it for 5.x and add upgrading info

**Review by escopecz — APPROVED** [review]
Thanks! 👍

**Pull request merged**